java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -82,4 +97,88 @@ private static boolean isBigIntegerPotentiallyBiggerThanLong(NewClassTree newCla
       return true;
     }
   }
+
+  private void reportWithQuickFix(NewClassTree newClassTree) {
+    QuickFixHelper.newIssue(context)
+      .forRule(this)
+      .onTree(newClassTree.identifier())
+      .withMessage("Remove this \"" + newClassTree.symbolType().name() + "\" constructor")
+      .withQuickFix(() -> computeQuickFix(newClassTree))
+      .report();
+  }
+
+  private static JavaQuickFix computeQuickFix(NewClassTree newClassTree) {
+    String message = "";
+    JavaTextEdit textEdit = null;
+    if (EMPTY_STRING_MATCHER.matches(newClassTree)) {
+      message = "Replace this \"String\" constructor with an empty string \"\"";
+      textEdit = JavaTextEdit.replaceTree(newClassTree, "\"\"");
+    } else if (BIG_INT_MATCHER.matches(newClassTree)) {
+      String arg = getFirstArgumentAsString(newClassTree).replace("\"", "");
+      try {
+        Integer.valueOf(arg);
+      } catch (NumberFormatException e) {
+        arg += "L";
+      }
+      String replacement = String.format("BigInteger.valueOf(%s)", arg);
```

Comment:
Because big integers can be as big as the biggest integer array that Java can support, the best you can do if you want to validate that number is to make sure the string matches `^(\+|\-)?[:digit:]+`